src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1230,7 +1230,7 @@ private BoundExpression BindSizeOf(SizeOfExpressionSyntax node, DiagnosticBag di
 
             BoundTypeExpression boundType = new BoundTypeExpression(typeSyntax, alias, typeWithAnnotations, typeHasErrors);
             ConstantValue constantValue = GetConstantSizeOf(type);
-            bool hasErrors = ReferenceEquals(constantValue, null) && ReportUnsafeIfNotAllowed(node, diagnostics, type);
+            bool hasErrors = constantValue is null && (type as NamedTypeSymbol)?.IsNativeInt != true && ReportUnsafeIfNotAllowed(node, diagnostics, type);
```

Comment:
>(type as NamedTypeSymbol)?.IsNativeInt != true [](start = 54, length = 46)

!(type is NamedTypeSymbol { IsNativeInt: true })

Reply:
Removed as it no longer applies.

---
In reply to: [334611757](https://github.com/dotnet/roslyn/pull/39157#discussion_r334611757) [](ancestors = 334611757)